app/helpers/common/priceFormatter.js

```diff
@@ -1,16 +1,15 @@
-export const formatPrice = (priceValue) => {
-  const formattedPrice = priceValue.toLocaleString(undefined, {
-    minimumFractionDigits: 3, maximumFractionDigits: 3,
-  });
+export const formatNumber = ({
+  value, minimumFractionDigits = 0, maximumFractionDigits = 4,
+}) => (
+  parseFloat(value)
+    .toLocaleString(
+      undefined, { minimumFractionDigits, maximumFractionDigits },
+    )
+);
 
-  const truncatedTo2dp = formattedPrice.substring(0, formattedPrice.indexOf('.') + 3);
-
-  return truncatedTo2dp;
-};
-
-export const formatDecimal = (priceValue) => {
-  if ((priceValue || {}).toString().includes('.') && priceValue.toString().split('.')[1].length < 3) {
-    return parseFloat(priceValue).toFixed(2);
+export const formatPrice = ({ value, minimumFractionDigits, maximumFractionDigits = 4 }) => {
```

Comment:
This implementation covers the case where `formatPrice({ value: 10.10, minimumFractionDigits: 0 })` has been called for example. In this case, it would fail the whole number check and `minimumFractionDigits: 2` would therefore be sent to `formatNumber`.

A default parameter `2` would get overriden by `0` and I would have to change it on a condition.

I should actually remove `minimumFractionDigits` within the if statement although the strict check for `0` ensures it never passes `undefined`.

Reply:
I'm not sure I follow.

In the example you give, `formatPrice({ value: 10.10, minimumFractionDigits: 0 })` `value` is not an integer so it will fail the whole number check and should be formatted to 2 d.p.

Reply:
Function now has proper default values, please take a look, I had to refactor a bit.

Reply:
I've had a closer look at this and I understand your point now.